### PR TITLE
Re-evaluate the execution order of the elif condition

### LINKED ISSUE #1147: The type of manifests may not be correctly determined when syncing content
```
http https://registry-1.docker.io/v2/lubosmj/test/manifests/latest Accept:application/vnd.docker.distribution.manifest.list.v2+json
```
```
HTTP/1.1 200 OK
content-length: 669
content-type: application/vnd.oci.image.index.v1+json
date: Fri, 25 Nov 2022 17:25:57 GMT
docker-content-digest: sha256:9a0a39570a12cf465209db9ab02812b02b87320bfcc1cee58978d1129191d0c8
docker-distribution-api-version: registry/2.0
docker-ratelimit-source: 86.49.248.80
etag: "sha256:9a0a39570a12cf465209db9ab02812b02b87320bfcc1cee58978d1129191d0c8"
ratelimit-limit: 100;w=21600
ratelimit-remaining: 91;w=21600
strict-transport-security: max-age=31536000

{
    "manifests": [
        {
            "digest": "sha256:d8fbbbf3fec1857c32c110292a9decf9744f9f97d7247019ae4776c241395221",
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 588
        },
        {
            "digest": "sha256:f8634bb68dccf0dc2a3113933a67f91dc10c4ac17dee90988cb6bc4ae55cf802",
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 588
        },
        {
            "digest": "sha256:6489ee892f64e59755435ee53f7d10cce5588a7788b4b2ae4a510a8bbc92704d",
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 588
        }
    ],
    "schemaVersion": 2
}

```

Error message:
```
pulp container repository sync --name test
Started background task /pulp/api/v3/tasks/cacfda02-b134-4fe5-9d7e-8ad23e9acfbc/
..........................................................................................Error: Task /pulp/api/v3/tasks/cacfda02-b134-4fe5-9d7e-8ad23e9acfbc/ failed: '{'errors': [{'code': ErrorDetail(string='MANIFEST_INVALID', code='parse_error'), 'message': ErrorDetail(string=": 'mediaType' is a required property", code='parse_error'), 'detail': {'digest': ErrorDetail(string='9a0a39570a12cf465209db9ab02812b02b87320bfcc1cee58978d1129191d0c8', code='parse_error')}}]}'
```

~~We should make the `mediaType` field optional like we already did for OCI indexes.~~

The function for determining media types of manifests is not working properly.

### PR DESCRIPTION
**Pull request by lubosmj**
closes #1147

### CODE REVIEW

File: pulp_container/app/utils.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def determine_media_type(content_data, response):
         # translate v1 signed media_type
         if media_type == MEDIA_TYPE.MANIFEST_V1_SIGNED:
             return MEDIA_TYPE.MANIFEST_V1
-        elif media_type in (MANIFEST_MEDIA_TYPES.IMAGE or MANIFEST_MEDIA_TYPES.LIST):
+        elif media_type in MANIFEST_MEDIA_TYPES.IMAGE or media_type in MANIFEST_MEDIA_TYPES.LIST:
             return media_type
```
Comment (ipanova) [inline pulp_container/app/utils.py:154]: does this need a data repair command/migration?

Comment (lubosmj) [reply pulp_container/app/utils.py:154]: We were basically skipping the fixed `elif` condition and passing the evaluation to https://github.com/pulp/pulp_container/blob/0b979a73676116ae959d6a16e01d61616b1730ac/pulp_container/app/utils.py#L164-L170. I think we are fine.

File: pulp_container/app/utils.py [hunk 2]
```diff
@@ -209,4 +209,6 @@ def validate_manifest(content_data, media_type, digest):
         validate(content_data, schema_validator)
     except ValidationError as error:
         # fail on the first encountered error
-        raise ManifestInvalid(reason=f'{".".join(error.path)}: {error.message}', digest=digest)
+        raise ManifestInvalid(
+            reason=f'{".".join(map(str, error.path))}: {error.message}', digest=digest
```
Comment (lubosmj) [inline pulp_container/app/utils.py:213]: This change was necessary to deal with a case where `error.path` contains the index of the affected value in a list of values (e.g., `deque(['manifests', 0, 'mediaType'])` is read as "the 0(1st)th entry in a list of manifests contains invalid mediaType").

Comment (ipanova) [reply pulp_container/app/utils.py:213]: I did not understand this comment. Can you give a specific example?

Comment (lubosmj) [reply pulp_container/app/utils.py:213]: Before:
```
Started background task /pulp/api/v3/tasks/7528dc91-9719-44fe-88e9-3810f53cb444/
................Error: Task /pulp/api/v3/tasks/7528dc91-9719-44fe-88e9-3810f53cb444/ failed: 'sequence item 1: expected str instance, int found'
```

After:
```
Started background task /pulp/api/v3/tasks/55a60363-d495-4594-b80a-3100d56b8439/
....Error: Task /pulp/api/v3/tasks/55a60363-d495-4594-b80a-3100d56b8439/ failed: '{'errors': [{'code': ErrorDetail(string='MANIFEST_INVALID', code='parse_error'), 'message': ErrorDetail(string="manifests.0.mediaType: 'application/vnd.docker.distribution.manifest.v2+json' is not one of ['application/vnd.oci.image.manifest.v1+json', 'application/vnd.oci.image.index.v1+json']", code='parse_error'), 'detail': {'digest': ErrorDetail(string='9a0a39570a12cf465209db9ab02812b02b87320bfcc1cee58978d1129191d0c8', code='parse_error')}}]}'
```

**Review by ipanova — APPROVED** [review]

**Pull request merged**